### PR TITLE
Mention Ellipsis pickling in the docs

### PR DESCRIPTION
**Pull request by sterliakov**
I literally came to docs to check if `Ellipsis` can be pickled and unpickled, but had to try it in REPL instead. This micro PR updates the documentation page to mention that `Ellipsis` is picklable. (no issue code, since this is a minor doc-only change)

Needs backport to all maintained versions, because this behaviour did not change (at least it's picklable in versions 3.7 to 3.12, I haven't checked earlier versions).

### CODE REVIEW

**Comment by AA-Turner** [comment]
Is this covered by ‘classes accessible from the top level of a module’? For example, `NotImplemented` isn’t contained in this list. Perhaps the ‘special’ thing about `Ellipsis` is the relation to `...`?

A

**Comment by sterliakov** [comment]
Oh, I'd rather add `NotImplemented` too, thanks for the reminder! Just checked, and it's picklable as well.

Neither `Ellipsis` nor `NotImplemented` are covered by "classes accessible from..." at least because they are not classes... These five builtin singletons (with None, True, and False) are special due to their nature. In fact `...` and `NotImplemented` are covered by "instances of such classes whose the result of calling `__getstate__()` is picklable", but this finding requires treating them in slightly unusual way (`Ellipsis` is not "instance of type `types.EllipsisType`" from the developer point of view, but a builtin truthy literal singleton, and the same logic applies to `NotImplemented` - instance of `types.NotImplementedType`).

File: Doc/library/pickle.rst [hunk 1]
```diff
@@ -494,7 +494,7 @@ What can be pickled and unpickled?
 
 The following types can be pickled:
 
-* ``None``, ``True``, and ``False``;
+* ``None``, ``Ellipsis``, ``True``, and ``False``;
```
Comment (AA-Turner) [inline Doc/library/pickle.rst]: As we now include more items than the 'obvious' None/True/False, give the group a name, and order by likely relative usage (e.g. put Ellipsis and NotImplemented last)

```suggestion
* built-in constants (``None``, ``True``, ``False``, ``Ellipsis``, and
  ``NotImplemented``);
```

**Comment by AA-Turner** [comment]
> Oh, I'd rather add `NotImplemented` too, thanks for the reminder!

Just added a review suggestion on that point.

Good points about classes vs singleton instances of the types!

A

File: Doc/library/pickle.rst [hunk 2]
```diff
@@ -494,7 +494,7 @@ What can be pickled and unpickled?
 
 The following types can be pickled:
 
-* ``None``, ``True``, and ``False``;
+* built-in constants (``None``, ``True``, ``False``, ``Ellipsis``, and ``NotImplemented``);
```
Comment (AA-Turner) [inline Doc/library/pickle.rst]: Wrap to 79 characters per earlier suggestion (some lines in this file don't, but it remains best practice).
```suggestion
* built-in constants (``None``, ``True``, ``False``, ``Ellipsis``, and
  ``NotImplemented``);
```

**Review by hauntsaninja — COMMENTED** [review]
If we're documenting this behaviour, ideally we should make sure it's tested as well.

**Comment by sterliakov** [comment]
Sorry for having abandoned this PR for a while. All these cases (both `NotImplemented` and `Ellipsis`) are already covered by tests introduced in [BPO6477](https://bugs.python.org/issue6477) 9 years ago together with `None` pickling test.

https://github.com/python/cpython/blob/d08d49dd09917531b113486976b03b5287b574f1/Lib/test/pickletester.py#L1976-L1983

**Review by hauntsaninja — APPROVED** [review]

**Comment by AA-Turner** [comment]
@hauntsaninja I've merged HEAD as the branch was out-of-date, so the checks would never succeed (& hence auto-merge would never happen)

A

**Pull request merged**